nesta/packages/meetup/meetup_utils.py

```diff
@@ -59,13 +64,56 @@ def flatten_data(list_json_data, keys, **kwargs):
                 else:
                     field_name = "_".join(k)
                     # Recursively assign the list of keys
-                    value = info                    
+                    value = info
                     for _k in k:
                         value = value[_k]
             # Ignore fields which aren't found (these will appear
             # as NULL in the database anyway)
             except KeyError:
                 continue
-            row[field_name] = value        
+            row[field_name] = value
         output.append(row)
     return output
+
+
+def get_members_by_percentile(engine, perc=10):
+    """Get the number of meetup group members for a given percentile
+    from the database.
+    
+    Args:
+        engine: A SQL alchemy connectable.
+        perc (int): A percentile to evaluate.
+    Returns:
+        members (float): The number of members corresponding to this percentile.
+    """
+    with db_session(engine) as session:
+        rows = (session
+                .query(Group.members)
+                .all())
+        rows = [r[0] for r in rows]
```

Comment:
If this query is very slow already then forget it, otherwise id probably get rid of that index:
```
        rows = (session
                .query(Group)
                .all())
        rows = [r.members for r in rows]
```
or maybe even just:
`rows = [g.members for g in session.query(Group).all()]`

It seems to be (slightly) slower than with the index

Reply:
It's not just slower, the memory requirement will also be fairly large since it will fetch the entire row (such as the description which is a `TEXT` field). Although your suggestion looks much prettier, I'd prefer to stick to the performant once since I've had `OutOfMemory` exceptions with meetup data in the past (especially if running on the wee batch machines!)

Reply:
indeed it does. I didn't think it executed the query until you actually interrogated a field within the object but it's actually `select *`'ing the whole row.... I have some serious optimisation to go back and do in my tasks...

Reply:
the `all()` does the execution. Doing a `fetchXXX()` command would reduce the memory overhead but obviously requires more requests, so will be considerably slower. I think the syntax you choose should be use case dependent

Reply:
you should still be able to do `.members` on it in place of [0] as it is a NamedTuple

Reply:
not that there is much point here, but I have just discovered this, in case it is useful when querying multiple fields

Reply:
ah - interesting, didn't know that. I'll put that in. Thanks!